src/providers/consensus/client.py

```diff
@@ -38,6 +40,20 @@ def get_block_root(self, state_id: Union[str, SlotNumber, BlockRoot]) -> BlockRo
         data, _ = self._get(self.API_GET_BLOCK_ROOT.format(state_id))
         return BlockRootResponse(**data)
 
+    def get_block_header(self, state_id: Union[str, SlotNumber, BlockRoot]) -> BlockHeaderFullResponse:
+        """
+        Spec: https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockRoot
+
+        No cache because this method is using to get finalized and head block, and they could not be cached by args.
+        """
+        data, rest_response = self._get(self.API_GET_BLOCK_HEADER.format(state_id))
+        resp = BlockHeaderFullResponse(data=BlockHeaderResponseData(**data), **rest_response)
+        if hasattr(resp, 'finalized') and not resp.finalized:
```

Comment:
Why here is such check?

Reply:
Perhaps I should move it. I think we’ll want to get some unfinalized slots somewhere

Reply:
fixed